### PR TITLE
[Fix-8337][Master] Process instance can not be kill when task is failure and can be retry

### LINKED ISSUE #8337: [Bug] [Master] Process instance can not be kill when task is failure and can be retry
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Process instance can not be kill when task is failure and can be retry

### What you expected to happen

stop successfully.

### How to reproduce

1.Create a process instance and a shell script that contains content ```exit 1``` and retry times is 3.
2.Running process instance.
3.Kill process instance while process instance in retry delay time.

### Anything else

_No response_

### Version

2.0.3

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

This pr will close #8337

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 1]
```diff
@@ -1437,14 +1444,10 @@ private ExecutionStatus processReadyPause() {
     private ExecutionStatus getProcessInstanceState(ProcessInstance instance) {
         ExecutionStatus state = instance.getState();
 
-        if (activeTaskProcessorMaps.size() > 0 || hasRetryTaskInStandBy()) {
+        if ((activeTaskProcessorMaps.size() > 0 || hasRetryTaskInStandBy()) && state != ExecutionStatus.READY_STOP) {
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: Is it better to put the process instance state check before?

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 2]
```diff
@@ -1615,6 +1645,9 @@ private void killAllTasks() {
             }
             TaskInstance taskInstance = processService.findTaskInstanceById(taskInstanceId);
             if (taskInstance == null || taskInstance.getState().typeIsFinished()) {
+                if (readyToSubmitTaskQueue.size() > 0) {
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: Why to clear submit task queue when some taskInstance is null or finish?

Comment (SbloodyS) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: When the task instance fails and the kill command is received in the task retry interval. We need to clear the readyToSubmitTaskQueue to ensure that the subsequent checks are normal.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: This has been check in readyToSubmitTaskQueue.clear(). I'll remove this logic.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Comment by caishunfeng** [comment]
PTAL @lenboo

**Pull request merged**